validator/project_validator.go

```diff
@@ -1898,3 +1899,25 @@ func parseS3PullParameters(c model.PluginCommandConf) (task, bv string, err erro
 	}
 	return task, bv, nil
 }
+
+// checkTaskTimeout checks if all tasks contain a timeout
+func checkTaskTimeout(project *model.Project) ValidationErrors {
+	errs := ValidationErrors{}
+	if project.ExecTimeoutSecs > 0 {
+		return errs
+	}
+	for _, task := range project.Tasks {
+		if task.ExecTimeoutSecs == 0 {
+			errs = append(errs,
+				ValidationError{
+					Message: fmt.Sprintf("task '%s' in project '%s' does not "+
```

Comment:
To avoid clutter we should just return a single warning if a config has any tasks without exec timeout secs defined, similar to how this message handles it: https://github.com/evergreen-ci/evergreen/blob/main/validator/project_validator.go#L567 Once we find a task without timeout secs defined we can append this warning and break the loop.